eva-server/src/main/java/uk/ac/ebi/eva/server/ws/GeneWSServerV2.java

```diff
@@ -70,80 +71,134 @@ public ResponseEntity getVariantsByGene(
+            @ApiParam(value = "Retrieve only variants whose Minor Allele Frequency is less than (<), less" +
+                    " than or equals (<=), greater than (>), greater than or equals (>=) or equals (=) the" +
+                    " provided number. e.g. <0.1")
+            @RequestParam(name = "maf", required = false) String maf,
+            @ApiParam(value = "Retrieve only variants whose PolyPhen score as stated by Ensembl VEP is less than" +
+                    " (<), less than or equals (<=), greater than (>), greater than or equals (>=) or equals (=) " +
+                    "the provided number. e.g. <0.1")
+            @RequestParam(name = "polyphen", required = false) String polyphenScore,
+            @ApiParam(value = "Retrieve only variants whose SIFT score as stated by Ensembl VEP is less than (<)," +
+                    " less than or equals (<=), greater than (>), greater than or equals (>=) or equals (=) the " +
+                    "provided number. e.g. <0.1")
+            @RequestParam(name = "sift", required = false) String siftScore,
+            @ApiParam(value = "Ensembl VEP release whose annotations will be included in the response, e.g. 78")
+            @RequestParam(name = "annot-vep-version", required = false) String
+                    annotationVepVersion,
+            @ApiParam(value = "Ensembl VEP cache release whose annotations will be included in the response, " +
+                    "e.g. 78")
+            @RequestParam(name = "annot-vep-cache-version", required = false) String
+                    annotationVepCacheVersion,
             @ApiParam(value = "The number of the page that should be displayed. Starts from 0 and is an integer.")
             @RequestParam(required = false, defaultValue = "0") Integer pageNumber,
             @ApiParam(value = "The number of elements that should be retrieved per page.")
             @RequestParam(required = false, defaultValue = "20") Integer pageSize,
+            @RequestParam(required = false, defaultValue = "0", name = "buffer") Integer bufferValue,
             HttpServletResponse response,
             @ApiIgnore HttpServletRequest request)
             throws IllegalArgumentException {
-        checkParameters(species, assembly);
+        checkParameters(species, assembly, bufferValue);
         MultiMongoDbFactory.setDatabaseNameForCurrentThread(DBAdaptorConnector.getDBName(species + "_" + assembly));
         List<FeatureCoordinates> featureCoordinates = service.findAllByGeneIdsOrGeneNames(geneIds, geneIds);
+
         if (featureCoordinates.isEmpty()) {
             return new ResponseEntity(featureCoordinates, HttpStatus.NO_CONTENT);
         }
 
+        if (bufferValue != 0) {
+            List<FeatureCoordinates> bufferCoordinates = new ArrayList<>();
+            featureCoordinates.forEach(coordinate -> {
+                bufferCoordinates.add(new FeatureCoordinates(null, null, null, coordinate.getChromosome(),
```

Comment:
this is ok, but just so you now, you could do 
```
List<FeatureCoordinates> bufferCoordinates = featureCoordinates.stream()
                  .map(new FeatureCoordinates(null, null, null, coordinate.getChromosome(),
                        coordinate.getStart() - bufferValue >= 0 ? coordinate.getStart() - bufferValue : 0,
                        coordinate.getEnd() + bufferValue));
                  .collect(Collectors.toList());
```

Reply:
Changed 👍